### PR TITLE
feat(Kafka connect ui): add connect ui command

### PR DESCRIPTION
**Pull request by Walesango2**
## Purpose
The purpose of this feature is to implement Kafka connect ui that will allow us to have access to see connector config and interact with the API graphically, without having to get onto the command line.

#### Linked Issues to Close
https://qmacbis.atlassian.net/browse/OY2-23716

## Approach
Added connect-ui line to run connect. 
Updated Kafka connect run command to enable connect ui to connect 
Connect-UI could not connect due to the restrictive binding of Kafka Connect. Jon assisted, and Simplified the connect command and bind to 0.0.0.0 so that Kafka connect is listening on all IPs.
We can now locally connect to the Kafka connect-ui

### CODE REVIEW

File: src/services/connector/serverless.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@ custom:
   brokerString: ${ssm:/aws/reference/secretsmanager/${self:custom.project}/${sls:stage}/brokerString, ssm:/aws/reference/secretsmanager/${self:custom.project}/default/brokerString}
   dbInfo: ${ssm:/aws/reference/secretsmanager/${self:custom.project}/${sls:stage}/dbInfo, ssm:/aws/reference/secretsmanager/${self:custom.project}/default/dbInfo}
   connectImage: ${ssm:/aws/reference/secretsmanager/ecr/images/${self:custom.project}/${self:service}, "confluentinc/cp-kafka-connect:6.0.9"}
+  connectUiImage: ${ssm:/aws/reference/secretsmanager/ecr/images/${self:custom.project}/${self:service}/connectui, "landoop/kafka-connect-ui"} # gettint hte container image for kafka connect ui
```
Comment (benjaminpaige) [inline src/services/connector/serverless.yml:60]: im not sure what "gettint hte container image for kafka connect ui" is suppoe to say buit we prob dont need the comment

File: src/services/connector/serverless.yml [hunk 2]
```diff
@@ -363,14 +404,35 @@ resources:
...
         Family: ${self:service}-${sls:stage}-kafka-connect-worker
         NetworkMode: awsvpc
         ExecutionRoleArn: !GetAtt KafkaConnectWorkerRole.Arn
         TaskRoleArn: !GetAtt KafkaConnectWorkerRole.Arn
         RequiresCompatibilities:
           - FARGATE
-        Memory: 4GB
-        Cpu: 2048
+        Memory: 8GB
+        Cpu: 4096
```
[the remark this answers is not in the document]
Comment (Walesango2) [reply src/services/connector/serverless.yml:435]: Yes, that is correct.

File: src/services/connector/serverless.yml [hunk 3]
```diff
@@ -225,11 +233,26 @@ resources:
...
+    KafkaConnectUiWorkerLogGroup: # created a new log group for  the kafka connect ui
+      Type: "AWS::Logs::LogGroup"
+      Properties:
+        LogGroupName: /aws/fargate/${self:service}-${sls:stage}-kafka-connect-ui
     KafkaConnectWorkerSecurityGroup:
       Type: AWS::EC2::SecurityGroup
       Properties:
         GroupDescription: Security Group for the Fargate Connect Workers.
         VpcId: ${self:custom.vpc.id}
+    KafkaConnectUiWorkerSecurityGroup: # created a secourity group for kafka connect ui
```
[the remark this answers is not in the document]
Comment (Walesango2) [reply src/services/connector/serverless.yml:245]: I will remove the comment.

File: src/services/connector/serverless.yml [hunk 4]
```diff
@@ -57,6 +57,8 @@ custom:
   brokerString: ${ssm:/aws/reference/secretsmanager/${self:custom.project}/${sls:stage}/brokerString, ssm:/aws/reference/secretsmanager/${self:custom.project}/default/brokerString}
   dbInfo: ${ssm:/aws/reference/secretsmanager/${self:custom.project}/${sls:stage}/dbInfo, ssm:/aws/reference/secretsmanager/${self:custom.project}/default/dbInfo}
   connectImage: ${ssm:/aws/reference/secretsmanager/ecr/images/${self:custom.project}/${self:service}, "confluentinc/cp-kafka-connect:6.0.9"}
+  connectUiImage: ${ssm:/aws/reference/secretsmanager/ecr/images/${self:custom.project}/${self:service}/connectui, "landoop/kafka-connect-ui"} # gettint hte container image for kafka connect ui
+  bigmacRoleArn: ${ssm:/aws/reference/secretsmanager/${self:custom.project}/${sls:stage}/bigmacRoleArn, ssm:/aws/reference/secretsmanager/${self:custom.project}/default/bigmacRoleArn}
```
[the remark this answers is not in the document]
Comment (Walesango2) [reply src/services/connector/serverless.yml:61]: connectUiImage: This is getting the image for the kafka connect ui container definition 
bigmacRoleArn is not being used

File: src/services/connector/serverless.yml [hunk 5]
```diff
@@ -225,11 +233,26 @@ resources:
...
+        LogGroupName: /aws/fargate/${self:service}-${sls:stage}-kafka-connect-ui
     KafkaConnectWorkerSecurityGroup:
       Type: AWS::EC2::SecurityGroup
       Properties:
         GroupDescription: Security Group for the Fargate Connect Workers.
         VpcId: ${self:custom.vpc.id}
+    KafkaConnectUiWorkerSecurityGroup: # created a secourity group for kafka connect ui
+      Type: AWS::EC2::SecurityGroup
+      Properties:
+        GroupDescription: Security Group for the Fargate Connect Ui Workers.
```
Comment (jdinh8124) [inline src/services/connector/serverless.yml:248]: Same with my other comment, this one triggered the security hub issue "Unused EC2 security groups should be removed" as well. I'm wondering if it initially starts off not being used unless its manually triggered via the run command on the cli

File: src/services/connector/serverless.yml [hunk 6]
```diff
@@ -74,15 +75,16 @@ custom:
...
           To forward the connector's Kafka Connect REST API to your localhost:8083, run:
           ecs-exec-pf -c ${self:service}-${sls:stage}-connect -t ${task##*/} -p 8083 -l 8083
-
-          To conect to the connector, run:
+          To forward the connector's Kafka Connect Ui to your localhost:8000, run:
+          ecs-exec-pf -c ${self:service}-${sls:stage}-connect -t ${task##*/} -p 8000 -l 8000
+          To connect to the connector, run:
           aws --region ${self:provider.region} ecs execute-command --cluster ${self:service}-${sls:stage}-connect --task ${task##*/} --container connect --interactive --command "/bin/sh"
-
+          To connect to the connector, run:
```
Comment (jdinh8124) [inline src/services/connector/serverless.yml:82]: Think maybe a different comment here

**Comment by benjaminpaige** [comment]
I like this. id love to see a demo. also i would ask that you could put together a page in the docs site that just lists out the steps to get this up a running. that would be sweet!!!

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**